homeassistant/components/simplisafe/__init__.py

```diff
@@ -450,7 +450,7 @@ async def async_init(self):
         """Initialize the data class."""
         asyncio.create_task(self.websocket.async_connect())
         self._hass.bus.async_listen_once(
-            EVENT_HOMEASSISTANT_STOP, self.websocket.async_disconnect
+            EVENT_HOMEASSISTANT_STOP, self.websocket.async_disconnect()
```

Comment:
The target function for the event should accept an event parameter.

Also, it's not good to use a coroutine object as a target since it may never be awaited if the event never fire, which may happen if the integration is unloaded before home assistant stops.

Reply:
@MartinHjelmare Meaning the target I have now is good (minus missing an event parameter)?

Reply:
No, the target now is a coroutine object. For target, we should create a function that accepts an event parameter and awaits `websocket.async_disconnect`.

Reply:
Ah, I see. Tired brain. I'll follow up with a new PR.

Reply:
Noticed now that we don't unregister this listener in `async_unload_entry`. We should do that.